File: backend/ScreenshotMonitor.Data/Entities/DeviceSecurityPolicy.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ScreenshotMonitor.Data.Entities;

public class DeviceSecurityPolicy
{
    [Key] public string Id { get; set; } = Guid.NewGuid().ToString();
    [Required, MaxLength(100)] public string DeviceId { get; set; } = string.Empty;
    [ForeignKey(nameof(DeviceId))] public Device Device { get; set; } = null!;
    public bool MonitoringEnabled { get; set; } = true;
    public bool ScreenshotsEnabled { get; set; } = true;
    public bool ActiveAppTrackingEnabled { get; set; } = true;
    public bool IdleTrackingEnabled { get; set; } = true;
    public bool BackupEnabled { get; set; } = true;
    public bool UsbAuditEnabled { get; set; } = true;
    public bool NetworkAuditEnabled { get; set; } = true;
    public bool FileChangeAuditEnabled { get; set; } = true;
    public bool AttendanceRemindersEnabled { get; set; } = true;
    public bool RestoreEnabled { get; set; } = true;
    [Required] public string UpdatedByAdminId { get; set; } = string.Empty;
    public DateTime UpdatedAt { get; set; }
}

File: frontend/desktop-tauri/src/deviceSecurityPolicy.js
export const defaultDeviceSecurityPolicy = Object.freeze({
  monitoringEnabled: true,
  screenshotsEnabled: true,
  activeAppTrackingEnabled: true,
  idleTrackingEnabled: true,
  backupEnabled: true,
  usbAuditEnabled: true,
  usbFileCopyAuditEnabled: true,
  networkAuditEnabled: true,
  fileChangeAuditEnabled: true,
  attendanceRemindersEnabled: true,
  restoreEnabled: true,
});

export function normalizeDeviceSecurityPolicy(value) {
  return Object.fromEntries(Object.entries(defaultDeviceSecurityPolicy)
    .map(([key, fallback]) => [key, typeof value?.[key] === 'boolean' ? value[key] : fallback]));
}

export async function loadDeviceSecurityPolicy({ request, storage }) {
  const deviceId = storage.getItem('screenMonitorDeviceId');
  if (!deviceId) return defaultDeviceSecurityPolicy;
  return normalizeDeviceSecurityPolicy(await request.get(`/security-policies/device/${deviceId}/effective`));
}

export function sameDeviceSecurityPolicy(left, right) {
  return Object.keys(defaultDeviceSecurityPolicy).every((key) => left?.[key] === right?.[key]);
}
